iac-extensions/docker/src/main/java/org/sonar/iac/docker/parser/DockerParser.java

```diff
@@ -74,4 +81,52 @@ private static void setParents(DockerTree tree) {
       setParents(child);
     }
   }
+
+  static class RecognitionExceptionAdjuster {
+    private static final String LINE_MESSAGE = "Parse error at line ";
+    private static final String COLUMN_MESSAGE = " column ";
+    private static final Pattern RECOGNITION_EXCEPTION_LINE_COLUMN_PATTERN = Pattern.compile(LINE_MESSAGE + "(?<line>\\d+)" + COLUMN_MESSAGE + "(?<column>\\d+)(?<rest>.*)");
+
+    private RecognitionExceptionAdjuster() {}
+
+    public static RecognitionException adjustLineAndColumnNumber(RecognitionException originalException, DockerPreprocessor.PreprocessorResult preprocessorResult) {
+      Matcher m = RECOGNITION_EXCEPTION_LINE_COLUMN_PATTERN.matcher(originalException.getMessage());
+      if (m.find()) {
+        int line = Integer.parseInt(m.group("line"));
+        int column = Integer.parseInt(m.group("column"));
+        String rest = m.group("rest");
+        int index = computeIndexFromLineAndColumn(preprocessorResult.processedSourceCode(), line, column);
+        int[] correctedLineAndColumn = preprocessorResult.sourceOffset().sourceLineAndColumnAt(index);
+        String newErrorMessage = buildNewErrorMessage(correctedLineAndColumn[0], correctedLineAndColumn[1], rest);
```

Comment:
We may want to change `RecognitionException` on SSLR side because it's build with a `TextRange` which would could perfectly use here.

Reply:
👍 Indeed that could be a nice improvement.
I see you assigned a task for yourself in the google doc "SSLR error reporting improvement", If you want you can assign this task to me and I will put the explanation directly.